examples/addmod.rs

```diff
@@ -0,0 +1,79 @@
+//! Addmod example for i64, i32, u64, u32.
+#![cfg_attr(target_arch = "wasm32", no_std)]
+#![cfg_attr(target_arch = "wasm32", no_main)]
+
+extern crate zink;
+use zink::primitives::numeric::Numeric;
+
+#[zink::external]
+pub fn addmod_i32(a: i32, b: i32, n: i32) -> i32 {
+    a.addmod(b, n)
+}
+
+#[zink::external]
+pub fn addmod_i64(a: i64, b: i64, n: i64) -> i64 {
+    a.addmod(b, n)
+}
+
+#[zink::external]
+pub fn addmod_u32(a: u32, b: u32, n: u32) -> u32 {
+    a.addmod(b, n)
+}
+
+#[zink::external]
+pub fn addmod_u64(a: u64, b: u64, n: u64) -> u64 {
+    a.addmod(b, n)
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn main() {}
+
+// FIXME: this test should pass, fix it to learn how to
+// add opcodes to the compiler
+#[ignore]
+#[test]
+fn test() -> anyhow::Result<()> {
+    use zint::{Bytes32 as _, Contract};
+
+    // Test for i32
+    let mut contract_i32 = Contract::search("addmod_i32")?.compile()?;
+    let info_i32 = contract_i32.execute([
+        "addmod_i32(uint32,uint32,uint32)".as_bytes(),
+        &3i32.to_bytes32(),
+        &5i32.to_bytes32(),
+        &7i32.to_bytes32(),
+    ])?;
+    assert_eq!(info_i32.ret, 1i32.to_bytes32());
+
+    // Test for i64
+    let mut contract_i64 = Contract::search("addmod_i64")?.compile()?;
+    let info_i64 = contract_i64.execute([
+        "addmod_i64(int64,int64,int64)".as_bytes(),
+        &3i64.to_bytes32(),
+        &5i64.to_bytes32(),
+        &7i64.to_bytes32(),
+    ])?;
+    assert_eq!(info_i64.ret, 1i64.to_bytes32());
```

Comment:
```suggestion
fn test() -> anyhow::Result<()> {
    use zint::{Bytes32 as _, Contract};

    // Test for i32
    let mut contract = Contract::search("addmod")?.compile()?;
    let info_i32 = contract.execute([
        "addmod_i32(int32,int32,int32)".as_bytes(),
        &3i32.to_bytes32(),
        &5i32.to_bytes32(),
        &7i32.to_bytes32(),
    ])?;
    assert_eq!(info_i32.ret, 1i32.to_bytes32());

    // Test for i64
    let info_i64 = contract.execute([
        "addmod_i64(int64,int64,int64)".as_bytes(),
        &3i64.to_bytes32(),
        &5i64.to_bytes32(),
        &7i64.to_bytes32(),
    ])?;
    assert_eq!(info_i64.ret, 1i64.to_bytes32());
```

Reply:
- we need to search contract `addmod` which is the name of the current example, all functions in this file are in the same contract
- once we have the contract instance, we don't need to re-compile the contract anymore